DataDog/datadog-agent
# [tagger] fix tag lookup when replay tagger is active

### What does this PR do?

This PR fixes bad tag enrichment for live traffic when a capture replay tagger state is loaded. Previously, a live traffic originID was only looked in the replay tagger state, where obviously originIDs would not  match. This PR makes sure we look into the live tagger state when no tags are found in the  loaded replay state.

### Motivation

Bug found during testing that reflected bad tagging of live traffic, and some extraneous side-effects.

### Additional Notes

Nope.

### Describe how to test your changes

To test this the ideal way is to:
1.  generate some traffic from a container (so you get an originID) that you will capture with an agent w/ origin detection enabled. Ideally you will tag this traffic with `traffic:capture` or something similar. 2. Send that traffic to the agents `statsd.sock` UDS socket and perform a `dogstatsd-capute` (a short 2-5m capture should be more than enough, and the packet rate can be arbitrary). 
3. Once the capture file is generated and the capture done, stop the `traffic:capture` generator.
4. Start a similar traffic generation process to target the dogstatsd server that submits the same metric name but tagged with `traffic:live`.
5. Replay the traffic we have just captured in step (3). 
6. You should now see in-app 2 (and only 2) distinct time series for the test metric name, and both properly tagged with the enriched tags for the originID. 

### Checklist


- [x] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.
